rts/System/SpringApp.cpp

```diff
@@ -144,6 +144,7 @@ DEFINE_string   (map,                                      "",    "Specify the m
 DEFINE_string   (menu,                                     "",    "Specify a lua menu archive to be used by spring");
 DEFINE_string   (name,                                     "",    "Set your player name");
 DEFINE_bool     (oldmenu,                                  false, "Start the old menu");
+DEFINE_string_EX(calc_checksum,      "calc-checksum",      "",    "Calculate named archive checksum");
```

Comment:
I've seen something on discord that this doesn't guarantee synchronicity. Maybe it would be good to put this in the description? (I guess this means it does affect archive cache after all)

Reply:
It runs exact same code as the full engine except without ThreadPool since it's not initialized that early, so it might run slower on SSDs. Generally my note was regarding running several instances of calc-checksum in parallel, which might certainly lead to troubles.

Reply:
Sounds good. It's not really documented what the real constraints on parallel engines are though[^1] and this one otherwise looks like a "const getter" so maybe worth mentioning anyway? 
```suggestion
DEFINE_string_EX(calc_checksum,      "calc-checksum",      "",    "Calculate named archive checksum and write to cache, cant run in parallel");
```

[^1]: somebody on ZK has a replay parser and he runs it in parallel and it seems to work pretty much fine, so maybe it's only dangerous if you don't touch sensitive files like springsettings or archive cache?